text/0012-CLI-commands.md

```diff
@@ -0,0 +1,1367 @@
+
+```
+qri rename [flags] <dataset_reference>
+```
+
+### Examples
+
+```
+  rename a dataset named annual_pop to annual_population:
+  $ qri rename me/annual_pop me/annual_population
+```
+
+### Options
+
+```
+  -h, --help   help for rename
+```
+
+
+
+<a id='qri_render'></a>
+## qri render
+
+Execute a template against a dataset
+
+### Synopsis
+
+You can use html templates, formatted in the go/html template style, 
+to render visualizations from your dataset. These visualizations can be charts, 
+graphs, or just display your dataset in a different format.
+
+Use the `--output` flag to save the rendered html to a file.
+
+Use the `--template` flag to use a custom template. If no template is
+provided, Qri will render the dataset with a default template.
+
+```
+qri render [flags] <dataset_reference>
+```
+
+### Examples
+
+```
+  render a dataset called me/schools to the file school.html:
+  $ qri render -o=schools.html me/schools
+
+  render a dataset with a custom template:
+  $ qri render --template=template.html me/schools
+```
+
+### Options
+
+```
+  -a, --all               read all dataset entries (overrides limit, offest)
+  -h, --help              help for render
+  -l, --limit int         max number of records to read (default 50)
+  -s, --offset int        number of records to skip
+  -o, --output string     path to write output file
+  -t, --template string   path to template file
+```
+
+
+
+<a id='qri_save'></a>
+## qri save
+
+Save changes to a dataset
+
+### Synopsis
+
+
+Save is how you change a dataset, updating one or more of data, metadata, and 
+structure. You can also update your data via url. Every time you run save, 
+an entry is added to your dataset’s log (which you can see by running `qri log 
+<dataset_reference>`). Every time you save, you can provide a message about what 
+you changed and why. If you don’t provide a message 
+Qri will automatically generate one for you.
+
+When you make an update and save a dataset that you originally added from a different
+peer, the dataset gets renamed from `peers_name/dataset_name` to `my_name/dataset_name`.
+
+The `--message` and `--title` flags allow you to add a commit message and title 
+to the save.
+
+Use `--no-publish` to not publish this version of the dataset to the Qri registry.
+
+```
+qri save [flags] <dataset_reference>
+```
+
+### Examples
+
+```
+  save updated data to dataset annual_pop:
+  $ qri --body /path/to/data.csv me/annual_pop
+
+  save updated dataset (no data) to annual_pop:
+  $ qri --file /path/to/dataset.yaml me/annual_pop
+```
+
+### Options
+
+```
+      --body string       path to file or url of data to add as dataset contents
+  -f, --file string       dataset data file (yaml or json)
+  -h, --help              help for save
+  -m, --message string    commit message for save
+  -p, --no-publish        do not publish this dataset to the registry
+      --secrets strings   transform secrets as comma separated key,value,key,value,... sequence
```

Comment:
Same publish/no-publish debate. Just highlighting for consistency.

Reply:
Should be the opposite of what it is here. `--publish : publish this dataset to the registry`

Reply:
So, in the registry docs, it says we publish by default. In the code we don't publish by default.

Whatever it is should be consistent with --new and --save